core/ts.h

```diff
@@ -201,6 +207,9 @@ class TransitionSystem
   // maps next back to curr
   smt::UnorderedTermMap curr_map_;
 
+  smt::TermVec constraints_;  ///< constraints added via
```

Comment:
Just a suggestion: maybe you could add some of the description of this PR as comment before this line, which will be valuable documentation, like: "extra vector of terms to TransitionSystems that records constraints added to the transition relation." and "For a functional system, you could now rebuild trans by AND-ing together all the equalities from state_updates_ and the constraints in constraints_."

Reply:
Great idea! Made the change with e0286ad16b965de5c7adbed1e99243314cb8a0d8